### PR TITLE
Update binaryen

### PR DESCRIPTION
**Pull request by MaxGraey**
This binaryen version [contain](https://github.com/WebAssembly/binaryen/pull/2870) [several](https://github.com/WebAssembly/binaryen/pull/2945) [improvments](https://github.com/WebAssembly/binaryen/pull/2914)

- [x] I've read the contributing guidelines

### CODE REVIEW

**Comment by dcodeIO** [comment]
Are we blocked on the `anyref` to `externref` rename here? Related: https://github.com/AssemblyScript/assemblyscript/pull/1319

**Comment by MaxGraey** [comment]
I don't think so

File: tests/compiler/features/reference-types.optimized.wat [hunk 1]
```diff
@@ -1,25 +1,25 @@
...
- (import "console" "log" (func $~lib/bindings/console/log (param anyref)))
- (import "Reflect" "get" (func $~lib/bindings/Reflect/get (param anyref anyref) (result anyref)))
- (import "reference-types" "external" (func $features/reference-types/external (param anyref) (result anyref)))
+ (import "console" "log" (func $~lib/bindings/console/log (param externref)))
+ (import "Reflect" "get" (func $~lib/bindings/Reflect/get (param externref externref) (result externref)))
+ (import "reference-types" "external" (func $features/reference-types/external (param externref) (result externref)))
  (memory $0 1)
  (data (i32.const 1024) "6\00\00\00\01\00\00\00\01\00\00\006\00\00\00f\00e\00a\00t\00u\00r\00e\00s\00/\00r\00e\00f\00e\00r\00e\00n\00c\00e\00-\00t\00y\00p\00e\00s\00.\00t\00s")
- (global $features/reference-types/nullGlobal (mut anyref) (ref.null))
+ (global $features/reference-types/nullGlobal (mut externref) (ref.null))
```
Comment (dcodeIO) [inline tests/compiler/features/reference-types.optimized.wat:17]: Looking through the fixtures, this is one such occasion introducing a breaking change for text format. Here, `anyref` has already been renamed to `externref`, but it is used with `ref.null`, which is now invalid iiuc.

Comment (MaxGraey) [reply tests/compiler/features/reference-types.optimized.wat:17]: I guess we could just comment or remove this code parts for now. WDYT?

Comment (dcodeIO) [reply tests/compiler/features/reference-types.optimized.wat:17]: The issue here is that updating Binaryen makes this a breaking change, so my preference would be to tackle https://github.com/AssemblyScript/assemblyscript/pull/1381 first, even if in a preliminary way, before updating to a version of Binaryen with renamed `externref`. Hmm...

Comment (MaxGraey) [reply tests/compiler/features/reference-types.optimized.wat:17]: > my preference would be to tackle #1381 first

Did you mean #1319 ?

**Pull request closed without merging**